skyportal/plot.py

```diff
@@ -433,104 +432,94 @@ def annotate_spec(plot, spectra, lower, upper):
-            if item_width > width_remaining:
-                plot.add_layout(
-                    Legend(
-                        orientation=legend_orientation,
-                        items=current_legend_items,
-                        click_policy="hide",
-                        location="top_center",
-                        margin=3,
-                    ),
-                    "below",
-                )
-                width_remaining = width - 120
-                current_legend_items = [item]
-            else:
-                current_legend_items.append(item)
-                width_remaining -= item_width
-        # Add remaining
-        plot.add_layout(
-            Legend(
-                orientation=legend_orientation,
-                items=current_legend_items,
-                click_policy="hide",
-                location="top_center",
-                margin=3,
-            ),
-            "below",
-        )
-    else:
-        plot.add_layout(
-            Legend(
-                click_policy="hide",
-                items=legend_items,
-                location="top_center",
-            ),
-            legend_loc,
-        )
+    plot.add_layout(
+        Legend(
+            orientation="horizontal",
+            items=legend_items,
+            click_policy="hide",
+            location="top_center",
+            margin=3,
+        ),
+        "below",
+    )
 
 
-def make_clear_photometry_button(model_dict):
-    """Make a button to clear photometry.
+def get_photometry_button_callback(info, model_dict):
+    return CustomJS(
+        args={'info': info, 'model_dict': model_dict},
+        code="""
+        for (const [key, value] of Object.entries(model_dict)) {
+          const [filter, origin, extra] = key.split("~");
+          if (info['filters'].includes(filter) || info['origins'].includes(origin)) {
+            value.visible = true;
+          }
+        }
+        """,
+    )
+
+
+def make_hide_photometry_button(model_dict):
+    """Make a button to hide photometry on the photometry plot.
     Returns
     -------
     bokeh Button object
     """
-    button = Button(name="Clear Photometry", label="Clear Photometry", width=112)
-    callback_clear_photometry = CustomJS(
+    button = Button(
+        name="Hide All Photometry", label="Hide All Photometry", width_policy="min"
+    )
+    callback_hide_photometry = CustomJS(
         args={'model_dict': model_dict},
         code="""
         for (const [key, value] of Object.entries(model_dict)) {
             value.visible = false
         }
         """,
     )
-    button.js_on_click(callback_clear_photometry)
+    button.js_on_click(callback_hide_photometry)
     return button
 
 
-def make_add_all_photometry_button(model_dict):
-    """Make a button to add photometry.
+def make_show_all_photometry_button(model_dict):
+    """Make a button to show all photometry on the photometry plot.
```

Comment:
I would add the `Parameters` heading in all the docstrings. And add an empty line before each heading.

Reply:
fixed